python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -171,6 +172,37 @@ def __getitem__(self, key: slice) -> Buffer:
             raise ValueError("slice must be C-contiguous")
         return self._getitem(offset=start, size=stop - start)
 
+    def copy(self, deep: bool = True):
+        """
+        Return a copy of Buffer.
+
+        Parameters
+        ----------
+        deep : bool, default True
+            If True, returns a deep-copy of the underlying Buffer data.
+            If False, returns a shallow-copy of the Buffer pointing to
+            the same underlying data.
+
+        Returns
+        -------
+        Buffer
+        """
+        if not deep:
+            shallow_copy = Buffer.__new__(Buffer)
+            shallow_copy._ptr = self._ptr
+            shallow_copy._size = self._size
+            shallow_copy._owner = self._owner
+            return shallow_copy
```

Comment:
> Does `SpillableBuffer` need to implement a `copy()` method or can it just safely inherit `Buffer.copy`?

Good point, I think it will work if we create a shallow copy by slicing: 

```suggestion
            return self[:]
```